### ISSUE TITLE
How to select training fMRI data subset for transfer learning, and the missing prepare_imgs_transfer.py

### ISSUE THREAD

**Issue by Concarne2** [WeizmannVision/brainit-fmri#2]
Hi,

first of all, the data/scripts/transfer/prepare_imgs_transfer.py file seems to be missing from this repo, even though some README files mentions it to prepare image subsets for transfer learning settings. I would be grateful if you could add it.

Also, I was curious how the subset of training data fMRI were selected for the limited-data transfer learning experiments. The missing prepare_imgs_transfer.py file could answer this question, but for now it seems unclear how the training data were selected for the limited data transfer learning experiments (15 min, 30 min, 1 hour). 

I also have one question: in the paper, it is mentioned that the 15-min experiment uses 450 fMRI samples. Is this a simple typo, or did the 15-min experiment in the paper actually use 450 fMRI samples? To my understanding, in NSD each hour corresponds to 750 fMRI samples (30,000 samples / 40 hour), and this can get even less if you consider some of them are reserved for the test set (MindEye2 as a consequence seems to use ~675 samples for the 1 hour experiments).

Thanks for the great work!

**Comment by Concarne2** [comment]
I also have one follow-up question: how much unlabeled synthetic data was used for the limited data (1 hour) transfer experiments? Was the whole ~120k data used alongside the 1 hour real subject-specific data, or was a smaller subset of the unlabeled data used during traning?

**Comment by RomaBeliy** [comment]
I apologize for the late response.

I have uploaded a script to prepare the transfer learning data.

Transfer learning selection procedure:
For the 1-hour setting we use all samples from the first session (one session corresponds to one hour) that are not part of the test set (~700 samples); for the 30-minute setting we use the first 375 of these samples; and for the 15-minute setting we use the first 187 of these samples.

"15-min experiment uses 450 fMRI samples":
This is a mistake in the paper. Thank you for pointing it out. It will be corrected.

Usage of unlabeled data in the transfer learning setup:
During transfer learning, we sample measured and external data with a 1:3 ratio (i.e., for every measured fMRI sample, we include three predicted external samples). The external samples are randomly sampled from the external dataset. For the 30-minute setting, each epoch contains 375 measured samples and 1125 external samples. Not all external samples are necessarily used during training.

Hopefully this answers your questions. Please let me know if anything is still unclear.
